### DISCUSSION TITLE
Wagmi requiredNamespaces

### DISCUSSION THREAD

**Discussion by shan-c** [wevm/wagmi#2581] [Q&A]
Wagmi by default gives a session proposal with only eip155:1 in requiredNamespace. All other chains that we mention in  are passed in optionalNamespace. Why is this

**Comment by jxom** [comment]
Because if the wallet doesn’t support one of the non-default chains in your app, the connection flow will break. It’s a WalletConnect v2 behavior.

**Comment by mars-peng-lb** [comment]
We also encountered the same problem. If we use "wagmi" with "requiredNamespaces," there will be only one namespace, while other items will be under "optionalNamespaces." Unfortunately, most wallets do not support "optionalNamespaces," such as imToken wallet. However, if we directly use the web3Modal SDK, it will work fine because "requiredNamespaces" can accept multiple namespaces.
Like this:
`{"id":1687353344785756,"params":{"id":1687353344785756,"pairingTopic":"0b028e505243fd2bd6fb021143e7a7721843f7bd6b479c77b63fa7198dc4c6bc","expiry":1687353752,"requiredNamespaces":{"eip155":{"methods":["eth_sendTransaction","eth_signTransaction","eth_sign","personal_sign","eth_signTypedData"],"chains":["eip155:80001","eip155:5","eip155:420"],"events":["chainChanged","accountsChanged"]}},"optionalNamespaces":{},"relays":[{"protocol":"irn"}],"proposer":{"publicKey":"bbf12098b419899e8601da0b8f99eb4f47100618154bc883c037b1b63914ec17","metadata":{"description":"React App for WalletConnect","url":"https://react-app.walletconnect.com/","icons":["https://avatars.githubusercontent.com/u/37784886"],"name":"React App"}}},"context":{"verified":{"verifyUrl":"","validation":"VALID","origin":"https://react-app.walletconnect.com/"}}}`

So which one is correct?
Thank you.
@jxom
